extensions/typescript-language-features/web/webServer.ts

```diff
@@ -5,263 +5,305 @@
+			}
+			catch (e) {
+				logger.info(`Error fs.directoryExists for ${path}`)
+				logger(e)
+				return false
+			}
+		},
+		createDirectory(path: string): void {
+			try {
+				logger.info(`calling createDirectory on ${path} (as ${toResource(path)})`)
+				// TODO: FileType.Directory might be correct! (need to learn about vscode's FileSystem.stat)
+				fs.createDirectory(toResource(path))
+			}
+			catch (e) {
+				logger.info(`Error fs.createDirectory`)
+				logger(e)
+			}
 		},
-		directoryExists: returnFalse, // Module resolution
-		readDirectory: notImplemented, // Configured project, typing installer
-		getDirectories: () => [], // For automatic type reference directives
-		createDirectory: notImplemented, // compile On save
-		writeFile: notImplemented, // compile on save
-		resolvePath: identity, // Plugins
-		// realpath? // Module resolution, symlinks
-		// getModifiedTime // File watching
-		// createSHA256Hash // telemetry of the project
+		getExecutingFilePath(): string {
+			logger.info('calling getExecutingFilePath')
+			return '/' // TODO: Might be correct! Or it might be /scheme/authority. Or /typescript. Or /scheme/authority/typescript. Or dist/browser/typescript
+		},
+		getCurrentDirectory(): string {
+			return '/' // TODO: Might still need to be /scheme/authority
+		},
+		getDirectories(path: string): string[] {
+			try {
+				logger.info('calling getDirectories on ' + path)
+				const entries = fs.readDirectory(toResource(path))
+				return entries.filter(([_, type]) => type === FileType.Directory).map(([f, _]) => f)
+			}
+			catch (e) {
+				logger.info(`Error fs.getDirectory`)
+				logger(e)
+				return []
+			}
+		},
+		/**
+		 * TODO: A lot of this code is made-up and should be copied from a known-good implementation
+		 * For example, I have NO idea how to easily support `depth`
+		 * Note: webServer.ts comments say this is used for configured project and typing installer.
+		 */
+		readDirectory(path: string, extensions?: readonly string[], exclude?: readonly string[]): string[] {
+			try {
+				logger.info('calling readDirectory on ' + path)
+				const entries = fs.readDirectory(toResource(path))
+				return entries
+					.filter(([f, type]) => type === FileType.File && (!extensions || extensions.some(ext => f.endsWith(ext))) && (!exclude || !exclude.includes(f)))
+					.map(([e, _]) => e)
+			}
+			catch (e) {
+				logger.info(`Error fs.readDirectory`)
+				logger(e)
+				return []
+			}
+		},
+		getModifiedTime(path: string): Date | undefined {
+			try {
+				logger.info('calling getModifiedTime on ' + path)
+				return new Date(fs.stat(toResource(path)).mtime)
+			}
+			catch (e) {
+				logger.info(`Error fs.getModifiedTime`)
+				logger(e)
+				return undefined
+			}
+		},
+		setModifiedTime(path: string): void {
+			logger.info('calling setModifiedTime on ' + path)
+			// But I don't have any idea of how to set the modified time to an arbitrary date!
```

Comment:
I don't think this is something VS Code's apis support

Reply:
Turns out that method is optional in the tsserver API and deleting it doesn't seem to confuse tsserver.